msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/common/algorithms/CurveTool.java

```diff
@@ -28,117 +28,125 @@
-			int matrixIndex = 0;
-			
-			
-			if(countTotalIteration>objSliceSparseMatrix.getSizeOfRawDataFile()){
-				System.out.println(countTotalIteration);
-				throw new IllegalArgumentException("Cannot calculate FWHM.");}
-			
-			Random generator = new Random(); 
-			int randInt = generator.nextInt(objSliceSparseMatrix.getSizeOfRawDataFile());
-			List<SliceSparseMatrix.VerticalSliceDataPoint> verticalSlice = objSliceSparseMatrix.getVerticalSlice(randInt);
-			
-			if(verticalSlice==null)
-	    		continue;
-			
-	    	WeightedObservedPoints obs = new WeightedObservedPoints();
-	    	
-	    	for (SliceSparseMatrix.VerticalSliceDataPoint datapoint:verticalSlice) {
-	    	   obs.add(datapoint.mz,datapoint.intensity);	    	  
-	    	   matrixIndex++;
-	    	}
-	    	
-	    	try{
-	    		double[] parameters = GaussianCurveFitter.create().fit(obs.toList());
-//	    		createDataFile(verticalSlice,parameters,randInt);
-		    	sigma  +=  2.35482*parameters[2];
-		    	
-	    	}
-	    	catch(Exception e){
-	    		continue;
-	    	}
-	    	countProperIteration++;
-		}
-		double fwhm = sigma/numberOfScansForFWHMCalc;
-		return fwhm;
-	}
-	
-	/**
-	 * <p>
-	 * roundFWHM method rounds the values of FWHM.This is Full width half maximum.
-	 * 
-	 * @param fwhm a {@link java.lang.Double} object.
-	 * 
-	 * @return roundedFWHM a {@link java.lang.Integer} object. This value is rounded integer value of fwhm.
-	 * </p>
-	 */
-	private int roundFWHM(double fwhm){
-		int roundedFWHM = (int) Math.round(fwhm+0.5);
-		return roundedFWHM;
-	}
-	
-	/**
-	 * <p>
-	 * createDataFile method creates text file for Gaussian results.
-	 * 
-	 * @param datapoint a {@link io.github.msdk.featdet.ADAP3D.common.algorithms.SliceSparseMatrix.VerticalSliceDataPoint} list. 
-	 * This list contains m/z values and corresponding intensities. 
-	 * @param parameters a {@link java.lang.Double} array. This is array of Gaussian parameters.
-	 * @param scanNumber a {@link java.lang.Integer} object. This is random scan numbers for which we're finding gaussian fit.
-	 * </p>
-	 */
-	private void createDataFile(List<SliceSparseMatrix.VerticalSliceDataPoint> datapoint,double[] parameters,int scanNumber){
-		double gaussianY [] = new double[datapoint.size()];
-		String filename = "output"+ scanNumber +".txt";
-		try{
-			File outputFile = new File(TEST_DATA_PATH + filename);
-			StringBuffer buffer = new StringBuffer ();
-			FileWriter writer = new FileWriter(outputFile);
-			
-			for(int i=0;i<datapoint.size();i++){
-				gaussianY[i] = parameters[0]*Math.exp(-Math.pow((datapoint.get(i).mz-parameters[1]), 2)/(2*Math.pow(parameters[2],2)));
-				buffer.append(datapoint.get(i).mz).append(",").append(datapoint.get(i).intensity).append(",").append(gaussianY[i]).append("\r\n");
-				writer.write(buffer.toString());
-				buffer = new StringBuffer();
-			}
-			writer.close();
-		}
-		
-		catch(Exception e){
-			
-		}
-		
-	}
+
+  private static final String TEST_DATA_PATH = "src/test/resources/";
```

Comment:
I think it was mentioned before - let's use getClass.getClassLoader.getResource() instead of hard-coding the path. Hard-coding the path fails in some environments.